pkg/kubectl/cmd/create_job.go

```diff
@@ -37,110 +34,96 @@ var (
-	if len(args) == 0 {
+func createJob(f cmdutil.Factory, cmdOut, cmdErr io.Writer, cmd *cobra.Command, args []string, argsLenAtDash int) error {
+	if len(args) == 0 || argsLenAtDash == 0 {
 		return cmdutil.UsageErrorf(cmd, "NAME is required")
 	}
-	c.Name = args[0]
-
-	c.From = cmdutil.GetFlagString(cmd, "from")
-	c.Namespace, _, err = f.DefaultNamespace()
-	if err != nil {
-		return err
+	name := args[0]
+	image := cmdutil.GetFlagString(cmd, "image")
+	from := cmdutil.GetFlagString(cmd, "from")
+	if len(image) == 0 && len(from) == 0 {
+		return fmt.Errorf("--image or --from must be specified")
 	}
-
-	clientset, err := f.KubernetesClientSet()
-	if err != nil {
-		return err
+	if len(image) != 0 && len(from) != 0 {
+		return fmt.Errorf("--image or --from must be specified, not both")
 	}
-	c.Client = clientset.BatchV1()
-	c.Builder = f.NewBuilder()
-	c.Cmd = cmd
 
-	return nil
-}
-
-func (c *CreateJobOptions) RunCreateJob() error {
-	infos, err := c.Builder.
-		Unstructured().
-		NamespaceParam(c.Namespace).DefaultNamespace().
-		ResourceTypeOrNameArgs(false, c.From).
-		Flatten().
-		Latest().
-		Do().
-		Infos()
-	if err != nil {
-		return err
-	}
-	if len(infos) != 1 {
-		return fmt.Errorf("from must be an existing cronjob")
-	}
-	cronJob, ok := infos[0].AsVersioned().(*batchv1beta1.CronJob)
-	if !ok {
-		return fmt.Errorf("from must be an existing cronjob")
+	var jobTemplate *batchv1beta1.JobTemplateSpec
+	if len(from) != 0 {
+		namespace, _, err := f.DefaultNamespace()
+		if err != nil {
+			return err
+		}
+		infos, err := f.NewBuilder().
+			Unstructured().
+			NamespaceParam(namespace).DefaultNamespace().
+			ResourceTypeOrNameArgs(false, from).
+			Flatten().
+			Latest().
+			Do().
+			Infos()
+		if err != nil {
+			return err
+		}
+		if len(infos) != 1 {
+			return fmt.Errorf("from must be an existing cronjob")
+		}
+		cronJob, ok := infos[0].AsVersioned().(*batchv1beta1.CronJob)
+		if !ok {
+			return fmt.Errorf("from must be an existing cronjob")
+		}
+		jobTemplate = &cronJob.Spec.JobTemplate
 	}
 
-	return c.createJob(cronJob)
-}
-
-func (c *CreateJobOptions) createJob(cronJob *batchv1beta1.CronJob) error {
-	annotations := make(map[string]string)
-	annotations["cronjob.kubernetes.io/instantiate"] = "manual"
-	for k, v := range cronJob.Spec.JobTemplate.Annotations {
-		annotations[k] = v
-	}
-	jobToCreate := &batchv1.Job{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:        c.Name,
-			Namespace:   c.Namespace,
-			Annotations: annotations,
-			Labels:      cronJob.Spec.JobTemplate.Labels,
-		},
-		Spec: cronJob.Spec.JobTemplate.Spec,
+	var command []string
+	if len(args) > 1 {
+		command = args[1:]
 	}
 
-	job, err := c.Client.Jobs(c.Namespace).Create(jobToCreate)
-	if err != nil {
-		return fmt.Errorf("failed to create job: %v", err)
+	var generator kubectl.StructuredGenerator
+	switch generatorName := cmdutil.GetFlagString(cmd, "generator"); generatorName {
+	case cmdutil.JobV1GeneratorName:
```

Comment:
So you moved a generator and attempted to re-use it and we can see here than re-using it fails to achieve the goal of the create command.  Stop using the generator, write in the simple object you actually want to create, and I could go for this.